File: countprimes.js
//Objective is to count the number of primes up to 'n'.

let n = 10


//O(nlognlogn) solution that uses the Sieve of Eratosthenes

let primes = new Array(n).fill(true)
primes[0] = false
primes[1] = false

for (let i = 2; i * i < n; i++) {
    if (primes[i]) {
        //Mark all multiples if 'i' false
        for (let j = i; j * i < n; j++) {
            primes[i * j] = false
        } 
    }
}

return primes.filter(boolean => boolean).length

File: countprimes.java
//Java Solution

class Solution {
    public int countPrimes(int n) {
        if (n <= 1) {
            return 0;
        }
        
        boolean[] dp = new boolean[n];
        Arrays.fill(dp, true);
        dp[0] = false;
        dp[1] = false;
        
        for (int i = 2; i * i < n; i++) {
            if (dp[i] == true) {
                for (int j = i; j * i < n; j++) {
                    dp[i * j] = false;
                }
            }
        }
        
        int count = 0;
        for (boolean b : dp) {
            if (b == true) {
                count++;
            }
        }
        
        return count;
    }
}